sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/ConfigurationRepository.java

```diff
@@ -252,6 +252,10 @@ public void reloadConfiguration(Collection<CType> configTypes) throws ConfigUpda
 
     private void reloadConfiguration0(Collection<CType> configTypes) {
         final Map<CType, SecurityDynamicConfiguration<?>> loaded = getConfigurationsFromIndex(configTypes);
+        // invalidate cache to load new configuration
+        if (!configCache.isEmpty()) {
+            configCache.clear();
```

Comment:
I looked this [up](https://stackoverflow.com/questions/17827573/map-putall-overrides-or-adds): I think it will only override matching content for which there is already a key and now it is getting a new value. If there were an old key that was no longer going to exist it would still persist. I don't think it will matter here but may for some future action.

Reply:
I'll remove this. Added it as initially as an alternative to security plugin's invalidateAll(), but we don't need it

And this would actually clear other CTypes from cache too which were not updated, so yep definitely removing this